recipes/gdal/all/conanfile.py

```diff
@@ -0,0 +1,668 @@
+        args.append("--with-geos={}".format("yes" if self.options.with_geos else "no"))
+        args.append("--without-sfcgal") # TODO: to implement when sfcgal lib available
+        args.append("--with-qhull={}".format("yes" if self.options.with_qhull else "no"))
+        args.append("--without-opencl") # TODO: to implement when opencl-headers available (and also OpenCL lib?)
+        args.append("--without-freexl") # TODO: to implement when freexl lib available
+        args.append("--with-libjson-c={}".format(self._unix_path(self.deps_cpp_info["json-c"].rootpath))) # always required !
+        if self.options.without_pam:
+            args.append("--without-pam")
+        args.append("--without-poppler") # TODO: to implement when poppler lib available
+        args.append("--without-podofo") # TODO: to implement when podofo lib available
+        args.append("--without-pdfium") # TODO: to implement when pdfium lib available
+        args.append("--without-perl")
+        args.append("--without-python")
+        args.append("--without-java")
+        args.append("--without-hdfs")
+        if tools.Version(self.version) >= "3.0.0":
+            args.append("--without-tiledb") # TODO: to implement when tiledb lib available
+        args.append("--without-mdb")
+        args.append("--without-rasdaman") # TODO: to implement when rasdaman lib available
+        if tools.Version(self.version) >= "3.1.0":
+            args.append("--without-rdb") # commercial library
+        args.append("--without-armadillo") # TODO: to implement when armadillo lib available
+        args.append("--without-cryptopp") # TODO: to implement when cryptopp lib available
+        args.append("--with-crypto={}".format("yes" if self.options.with_crypto else "no"))
+        args.append("--with-lerc={}".format("no" if self.options.without_lerc else "yes"))
+        if self.options.with_null:
+            args.append("--with-null")
+        if self.options.get_safe("with_exr") is not None:
+            args.append("--with-exr={}".format("yes" if self.options.with_exr else "no"))
+
+        # Some tricks in LDFLAGS env variable
+        env_build_vars = self._autotools.vars
+        if self.settings.os == "Linux":
+            # Inject lib directories in rpath in case of shared dependencies
+            rpath = ""
+            for libdir in self.deps_cpp_info.libdirs:
+                rpath += " -Wl,-rpath=\"{}\"".format(libdir)
+            env_build_vars["LDFLAGS"] = env_build_vars["LDFLAGS"] + rpath
+            # Inject -stdlib=libc++ for clang with libc++
+            if self.settings.compiler == "clang" and self._stdcpp_library == "c++":
+                env_build_vars["LDFLAGS"] = "-stdlib=libc++ " + env_build_vars["LDFLAGS"]
+
+        with tools.chdir(configure_dir):
+            self._autotools.configure(args=args, vars=env_build_vars)
```

Comment:
This is a workaround for https://github.com/conan-io/conan/issues/7133 and https://github.com/conan-io/conan/issues/7109

Reply:
@danimtb @jgsogo @SSE4 @uilianries @Artalus
Should I use `-Wl,-rpath-link=${libdir}` instead of `-Wl,-rpath=${libdir}` to avoid hardcoding path in executables?
If it the case, there is an issue, because pkg-config generator adds `-Wl,-rpath=${libdir}` not `-Wl,-rpath-link=${libdir}`.

Reply:
Not an expert, so consider this an opinion rather than answer.
The generator (which works during `conan install`) is right in this, since the installed libraries are supposed to be used on this local machine. So `-rpath` instead of `-rpath-link` alllows everything to just work™ without LD_LIBRARY_PATH and such.
In `package()` you call `.install()`, which, in case of Autotools AFAIK, strips any RPATH entries from binaries built. So it seems to me that you are free to use any form of rpath flag.
You might also want to consider [`conans.client.build.compiler_flags.rpath_flags()`](https://github.com/conan-io/conan/blob/develop/conans/client/build/compiler_flags.py#L16)

Reply:
Maybe I found what's wrong in GDAL makefiles. If using libtool, gdal relies on libtool intradependencies, so it:
- uses $(LDFLAGS) and $(LIBS) in link command of gdal shared lib
- uses $(LDFLAGS) but not $(LIBS) in link command of gdal executables, since it links `libgdal.la` and relies on intradependencies mechanism of libtool. For some reason, it doesn't seem to always be reliable, but I'm not familiar with libtool.

Here is where this logic of not using $(LIBS) occurs:

GDALmake.opt.in
```makefile
# libtool targets and help variables
LIBGDAL	:=		libgdal.la
LIBGDAL_CURRENT	:=	27
LIBGDAL_REVISION	:=	0
LIBGDAL_AGE	:=	0

(...)

ifeq ($(HAVE_LIBTOOL), yes)

CONFIG_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL)
ifeq ($(MACOSX_FRAMEWORK),yes)
CONFIG_LIBS_INS	=	-L$(INST_LIB)/unix/lib -lgdal
else
CONFIG_LIBS_INS	=	-L$(INST_LIB) -lgdal
endif # MACOSX_FRAMEWORK

EXE_DEP_LIBS	=	$(GDAL_ROOT)/$(LIBGDAL)
SO_EXT		=	la

else # HAVE_LIBTOOL

ifeq ($(HAVE_LD_SHARED),yes)
CONFIG_LIBS	=	$(GDAL_SLIB_LINK) $(LIBS)
ifeq ($(MACOSX_FRAMEWORK),yes)
CONFIG_LIBS_INS	=	-L$(INST_LIB)/unix/lib -lgdal
else
CONFIG_LIBS_INS	=	-L$(INST_LIB) -lgdal
endif
EXE_DEP_LIBS	=	$(GDAL_SLIB)
else
CONFIG_LIBS	=	$(GDAL_LIBS) $(LIBS)
CONFIG_LIBS_INS	=	$(foreach LF,$(GDAL_LIBS),$(INST_LIB)/$(notdir $(LF)))\
			 $(LIBS)
EXE_DEP_LIBS	=	$(GDAL_LIB)
endif

endif # HAVE_LIBTOOL
```

Then in `gdal/apps/GNUmakefile`:

```makefile
gdalinfo$(EXE):	gdalinfo_bin.$(OBJ_EXT) $(DEP_LIBS)
	$(LD) $(LNK_FLAGS) $< $(XTRAOBJ) $(CONFIG_LIBS) $(CONFIG_LIB_UTILS) -o $@
```

In this command line, $(LIBS) is never added in any variable if using libtool.

So I can ever:
- patch `GDALmake.opt.in` to add $(LIBS) in $(CONFIG_LIBS) even if using libtool.
- other solution?

Reply:
I've checked and adding indirectly $(LIBS) in link commands that build gdal executables fixes this issue, no need of `-rpath` or `-rpath-link`.